source/agora/utils/PrettyPrinter.d

```diff
@@ -275,15 +283,17 @@ private struct TransactionFmt
             enum OutputPerLine = 3;
 
             if (this.value.inputs.length)
-                formattedWrite(sink, "Type : %s, Inputs (%d): %(%(%s, %),\n%)\n",
+                formattedWrite(sink, "Type : %s, Inputs (%d):%s%(%(%s, %),\n%)\n",
                     this.value.type,
                     this.value.inputs.length,
+                    startNewLineIfMultiple(InputPerLine, this.value.inputs.length),
                     this.value.inputs.map!(v => InputFmt(v)).chunks(InputPerLine));
             else
                 formattedWrite(sink, "Type : %s, Inputs: None\n", this.value.type);
 
-            formattedWrite(sink, "Outputs (%d): %(%(%s, %),\n%)",
+            formattedWrite(sink, "Outputs (%d):%s%(%(%s, %),\n%)",
                 this.value.outputs.length,
+                startNewLineIfMultiple(OutputPerLine, this.value.outputs.length),
```

Comment:
`this.value.outputs.length > OutputPerLine ? "\n" : " "`

Reply:
Updated